server/src/test/java/org/candlepin/policy/AutobindRulesTest.java

```diff
@@ -2420,7 +2431,7 @@ public void testShouldSelectAllNonStackedPoolsThatProvideRoleAddonOrProductButNo
 
         // This pool provides the consumer's installed product, but does not provide his specified role.
         Product prodWithInstalledProductOnly = createSysPurposeProduct(null, null, null,
```

Comment:
This test is a variation of the previous, but with the usage/sla/service_type pools that we expect to not be attached to have an installed product that the user does not specify on them, so we should do what we did in the previous test:
add service_type to the consumer, and also add a pool that provides that service_type and also a random installed product, and assert that it will not be attached

Reply:
Done.